### PR TITLE
Explore: Fix using data source line limit when opening logs sample in split view

### LINKED ISSUE #64038: Logs sample: Open logs in split view sets 100 log lines limit
**What happened**:
When user runs Loki metric query in Explore, opens `Log sample` and then clicks on `Open log in split view`, line limit of 100 (applied to log samples) is also applied to all queries in split view. This should not happen and we should get default log line limit. 

**How to reproduce it (as minimally and precisely as possible)**:

https://user-images.githubusercontent.com/30407135/222473236-fac17644-d131-48e3-ae3d-2a06e28a02db.mov

**Anything else we need to know?**:

**Environment**: 9.4/main

### PR DESCRIPTION
**Pull request by ivanahuckova**
In this PR, we are fixing line limit set for Loki queries when opening logs sample in split mode. Previously, 100 lines limit was set for log queries executed in split view, but in this PR we are fixing this, so the data source will use `maxLines` limit for these queries. 100 lines limit for logs sample stays the same.

Fixes https://github.com/grafana/grafana/issues/64038

**Special notes for your reviewer:**

https://user-images.githubusercontent.com/30407135/232081543-a8219909-1edb-4833-960e-41c8498e9f99.mov

### CODE REVIEW

**Comment by ivanahuckova** [comment]
I have removed `levitate breaking change` label as changes have been done only to `@internal` APIs.

File: packages/grafana-data/src/types/logs.ts [hunk 1]
```diff
@@ -163,6 +163,26 @@ export enum SupplementaryQueryType {
...
+ * @internal
+ */
+export type LogsVolumeOption = {
+  type: SupplementaryQueryType.LogsVolume;
+};
+
+/**
+ * @internal
+ */
+export type LogsSampleOptions = {
```
Comment (matyax) [inline packages/grafana-data/src/types/logs.ts:181]: Why did you mark these as internal? I think it's the first time I see this annotation used.

File: packages/grafana-data/src/types/logs.ts [hunk 2]
```diff
@@ -206,7 +226,7 @@ export interface DataSourceWithSupplementaryQueriesSupport<TQuery extends DataQu
    * Returns a supplementary query to be used to fetch supplementary data based on the provided type and original query.
    * If provided query is not suitable for provided supplementary query type, undefined should be returned.
    */
-  getSupplementaryQuery(type: SupplementaryQueryType, query: TQuery): TQuery | undefined;
+  getSupplementaryQuery(options: SupplementaryQueryOptions, originalQuery: TQuery): TQuery | undefined;
```
Comment (matyax) [inline packages/grafana-data/src/types/logs.ts:229]: Using `options` is a great idea.

**Review by matyax — APPROVED** [review]
Looks good!

**Pull request merged**